Update cnc-ddraw.yml

MikaMika opened the pull request:
New repo FunkyFr3sh/cnc-ddraw: previous from CnCNet/cnc-ddraw no longer exists. Updated from v4.4.8.0 to v7.0.0.0

Fixes #289

## Type of change
- [ ] New dependency
- [x] Manifest fix
- [ ] Other

# Was this tested using a [local repository](https://maintainers.usebottles.com/Testing)?
- [ ] Yes
- [x] No

MikaMika committed:
Update cnc-ddraw.yml

New repo FunkyFr3sh/cnc-ddraw: previous from CnCNet/cnc-ddraw no longer exists.
Updated from v4.4.8.0 to v7.0.0.0

MikaMika commented:
I chose v7.0 in favor of v7.1 as I encountered a black screen issue in StarCraft Classic with the most recent one. I'm filing an Issue there to get it fixed, then this dependency could be updated when v7.2 or v7.3 releases.

jntesteves reviewed (changes requested):
Thanks for this PR. It's looking good, except the shaders. I've looked into the new zip file, and there are new shaders that won't be matched by the glob `*.glsl` because they are named `.glsl.pass1` for a couple multi-pass shaders. I think all the globs should be changed to something like `*.*` or maybe even just `*` if that works because we really want to copy all files in these folders no matter what they are called.

Also, a couple shaders were removed in this release. The `cubic` and `windowed` folders don't exist anymore.

MikaMika committed:
Update cnc-ddraw.yml

MikaMika committed:
Update cnc-ddraw.yml

archive_extract returns an error code 2 with most zip provided by FunkyFr3sh/cnc-ddraw. Seems to be related to 'cnc-ddraw config.exe' being extracted as a directory and crashing 7z. A solution would be to append an argument to extract only a selection of files and avoid the one crashing, but I cannot figure out this action. As an alternative, cnc-ddraw-experimental-debuglog.zip does not contain the problematic exe and the extraction is successful, although without Shaders. The priority is still to provide a working dll for software requiring it, while Shaders are not necessary.

MikaMika commented:
Sorry for not testing using a local repo earlier: I thought it would be a simple url fix.
I have done the testing now and found out this: 
archive_extract returns an error code 2 with most zip provided by FunkyFr3sh/cnc-ddraw. Seems to be related to 'cnc-ddraw config.exe' being extracted as a directory and crashing 7z. A solution would be to append an argument to extract only a selection of files and avoid the one crashing, but I cannot figure out this action. As an alternative, cnc-ddraw-experimental-debuglog.zip does not contain the problematic exe and the extraction is successful, although without Shaders. The priority is still to provide a working dll for software requiring it, while Shaders are not necessary.

MikaMika committed:
Update cnc-ddraw.yml

MikaMika commented:
Sorry for more changes, I realized I made some mistakes:
* The cnc-ddraw-experimental-debuglog includes a FPS counter at the top-left corner, which should not be for user.
* Reverting to v7.0.0.0, as the 7z bug is due to the version 17.05 included in the Flatpak and should be reported there instead. I have reported to FunkyFr3sh the black screen bug in v7.1.0.0 and waiting for a fix.
* Copying all files with `*.*` instead of each file types. `*` will try copying directories and error.
* Copying only `win32/Shaders/interpolation/catmull-rom-bilinear.glsl` as this is the default one in `ddraw.ini`, all other shaders are available in `win32/Shaders/shader-package.zip`.

I'm preparing to file an issue for https://github.com/flathub/com.usebottles.bottles/blob/master/com.usebottles.bottles.yml to use another version of 7-zip, and the extract bug should be resolved.

sham1 commented:
Since the 7-Zip has been merged over in the Wine flatpak on Flathub for almost a year now, I'm wondering if this might go ahead. I would need to get cnc-ddraw working for some of the things I want to use Bottles with, and while I can install cnc-ddraw manually, it'd be nice to get this one working for easier management.

mirkobrombin committed:
Fix cnc-ddraw extraction path

mirkobrombin reviewed (approved):
Tested through the Bottles dependency manager with the current Flatpak runtime. The archive extracts and the DLL, configuration tool and required shaders are installed.
